src/NumberInput/NumberInput.svelte

```diff
@@ -201,6 +256,7 @@
         max="{max}"
         min="{min}"
         step="{step}"
+        on:input="{updatePreviousValue}"
```

Comment:
- Events should be at the end, after rest props.
- There already is an `input` event handler, the logic can be combined into/called from the existing `onInput` function; there should be no need to attach two different handlers.